src/Twig/Component/WishlistDetails/CollectiveActions/CopyToWishlistComponent.php

```diff
@@ -0,0 +1,146 @@
+use Symfony\Contracts\Translation\TranslatorInterface;
+use Symfony\UX\LiveComponent\Attribute\AsLiveComponent;
+use Symfony\UX\LiveComponent\Attribute\LiveAction;
+use Symfony\UX\LiveComponent\Attribute\LiveProp;
+use Symfony\UX\LiveComponent\ComponentToolsTrait;
+use Symfony\UX\LiveComponent\DefaultActionTrait;
+
+#[AsLiveComponent]
+final class CopyToWishlistComponent
+{
+    use ComponentToolsTrait;
+    use DefaultActionTrait;
+    use TemplatePropTrait;
+    use HookableLiveComponentTrait;
+
+    #[LiveProp]
+    public int $wishlistId;
+
+    /** @var array<int, array{name: string, id: int}> */
+    #[LiveProp]
+    public array $wishlists = [];
+
+    #[LiveProp]
+    public bool $showModal = false;
+
+    #[LiveProp(writable: true)]
+    public ?int $selectedId = null;
+
+    /** JSON string with preselected variant ids from main list */
+    #[LiveProp(writable: true)]
+    public string $preselected = '[]';
+
+    public function __construct(
+        private readonly WishlistRepositoryInterface $wishlistRepository,
+        private readonly MessageBusInterface $commandBus,
+        private readonly RequestStack $requestStack,
+        private readonly TranslatorInterface $translator,
+        private readonly UrlGeneratorInterface $urlGenerator,
+        private readonly WishlistsResolverInterface $wishlistsResolver,
+    ) {
+    }
+
+    #[LiveAction]
+    public function openModal(?string $selection = null): void
+    {
+        $this->showModal = true;
+        if ($this->wishlists === []) {
+            $resolved = $this->wishlistsResolver->resolveAndCreate();
+            foreach ($resolved as $w) {
+                $this->wishlists[] = ['id' => (int) $w->getId(), 'name' => (string) $w->getName()];
+            }
+        }
+        if (null !== $selection) {
+            $this->preselected = $selection;
+        }
+    }
+
+    #[LiveAction]
+    public function closeModal(): void
+    {
+        $this->showModal = false;
+        $this->selectedId = null;
+    }
+
+    #[LiveAction]
+    public function copy(?string $selection = null): RedirectResponse
+    {
+        /** @var Session $session */
+        $session = $this->requestStack->getSession();
+
+        if (null === $this->selectedId) {
+            return $this->redirectBack();
+        }
+
+        /** @var ?WishlistInterface $wishlist */
+        $wishlist = $this->wishlistRepository->find($this->wishlistId);
+        if (null === $wishlist) {
+            return $this->redirectBack();
+        }
+
+        $variantIds = [];
+        if (null !== $selection) {
+            $decoded = json_decode($selection, true);
+            if (\is_array($decoded)) {
+                $variantIds = array_map('intval', $decoded);
+            }
+        } else {
+            $decoded = json_decode($this->preselected, true);
+            if (\is_array($decoded)) {
+                $variantIds = array_map('intval', $decoded);
+            }
+        }
```

Comment:
I feel like we'll need a comment as for what is what here, we have 3 vars `selection`, `selectedId` and `preselected`; what is what and what are their responsibilities?